lib/richtext_controller.dart

```diff
@@ -0,0 +1,243 @@
+import 'dart:async';
+
+import 'package:enough_html_editor/enough_html_editor.dart';
+import 'package:flutter/material.dart' as ui;
+import 'package:flutter/material.dart';
+import 'package:rich_text_composer/views/widgets/rich_text_option_bottom_sheet.dart';
+
+import 'models/types.dart';
+
+class RichTextController {
+  HtmlEditorApi? htmlEditorApi;
+
+  final listSpecialTextStyleApply = ValueNotifier<Set<SpecialStyleType>>({});
+  final paragraphTypeApply =
+      ValueNotifier<ParagraphType>(ParagraphType.alignLeft);
+  final dentTypeApply = ValueNotifier<DentType?>(null);
+  final orderListTypeApply = ValueNotifier<OrderListType?>(null);
+  final selectedTextColor = ValueNotifier<ui.Color>(ui.Colors.black);
+  final selectedTextBackgroundColor = ValueNotifier<ui.Color>(ui.Colors.white);
+  final headerStyleTypeApply =
+      ValueNotifier<HeaderStyleType>(HeaderStyleType.normal);
+
+  final StreamController<bool> richTextStreamController =
+      StreamController<bool>();
+
+  bool isBoldStyleAppended = false;
+  bool isItalicStyleAppended = false;
+  bool isUnderlineAppended = false;
+  bool isStrikeThroughAppended = false;
+
+  Stream<bool> get richTextStream => richTextStreamController.stream;
+
+  void listenHtmlEditorApi() {
+    htmlEditorApi?.onFormatSettingsChanged = (formatSettings) {
+      if (formatSettings.isBold) {
+        isBoldStyleAppended = true;
+        listSpecialTextStyleApply.value.add(SpecialStyleType.bold);
+        listSpecialTextStyleApply.notifyListeners();
```

Comment:
IMO, you don't need to call it. The `ValueNotifier ` variable will dynamically call `notifyListeners ` when you change its value.


<img width="623" alt="Screen Shot 2022-09-30 at 13 04 40" src="https://user-images.githubusercontent.com/80730648/193201368-61330636-184c-4ed1-8a0f-5fb23787356e.png">

Reply:
@dab246 IMO, The `ValueNotifier` variable will dynamically call `notifyListeners` when you change its value.  But when I added a new object to the list, the list will not change so `notifyListeners` will not call.